### ISSUE TITLE
Church feature for Cleanliness

### ISSUE THREAD

**Issue by aldopain** [aldopain/corpwars#87]
Итак, идея unique фичи для фракции Чистота. 
Они поборники чистоты, религиозники, всё такое. 
Церковь:
Дает счастье _всем_ 3 слоям населения за удовлетворение потребностей церкви
Эффект улучшается, если:
1. Война с Развитием
2. Захвачены религиозные святыни

**Comment by fshpsmgc** [comment]
Кстати, им можно ещё ёбнуть отстутствие богачей как класса (или, скорее, очень лимитированное количество богачей, так как они все занимают высокие посты в церкви), и доход от богатого класса зависит от развития церкви

Плюс, я думаю, интересной фичей будет переписывание святых текстов под себя. Допустим, эти сто лет -- это время реформации их религии, и мы сможем переписывать её части для различных бонусов. 

Плюс, уникальные здания (возможно, взамен научным/развлекательным), которые дают бонус к религии

**Comment by aldopain** [comment]
Найс, да.

**Comment by danibord** [comment]
Первоначальное предложение звучит хорошо, ограничение/ избавление от класса богатых для данной фракции тоже кажется хорошей идеей. 

Но вот что значит "переписывание святых текстов под себя" с точки зрения механики ? Как это будет происходить в игре? Что по вашему представляют из себя религиозные тексты внутри игры?

**Comment by fshpsmgc** [comment]
Примерно как в Civ5
То есть, у нас есть религия со своим текстом (который было бы очень неплохо написать, хотя бы основные места), потом, каждые N лет (очевидно, не строгая цифра, а она должна зависеть от парочки параметров, вроде счастья населения и развития церкви) мы получаем возможность изменить один основополагающий принцип на один из трёх других
То есть, допустим, у нас убийство это смертный грех. Но мы хотим это изменить. У нас есть выбор
- убийство -- смертный грех, но только, если ты не воюешь против неверных (бонус к силе юнитов)
- убийство простительно, если ты (у меня даже на абстрактный пример кончилась фантазия) хорошо работаешь на производстве (бонус к производству ресурсов)
- убийство совсем не грех (штраф к счастью, бонус к доходу в городах)


Кстати, пока я писал этот пример, то мне пришла идея просто сделать так, что у нас есть 10-15 смертных грехов и мы __немножко__ изменяем их.

**Comment by danibord** [comment]
Т.е. грубо говоря у нас есть некоторые законы, соблюдая которые игрок будет получать некие плюшки. При этом с течением игры можно менять набор данных законов ?

**Comment by fshpsmgc** [comment]
Вообще, нет, это скорее набор пассивных бонусов, которые можно менять
То есть, твоя идея звучит поинтереснее, но это довольно большая фишка и я так с ходу не уверен, как мы её сбалансим (и даже реализуем)

**Comment by danibord** [comment]
ну, я лишь уточняю в каком виде Вы видите реализацию подобной идеи)

**Comment by aldopain** [comment]
А мы это и не реализуем, жанр не тот. 
Но вообще можно _почти_ так сделать, пилим принципы (чистота плоти, нет разврату), игрок строит заводы по производству дилдо, веруны его за это карают.

**Comment by aldopain** [comment]
Это не совсем то же самое, но близко.

**Comment by danibord** [comment]
Т.к. мы с тобой это вместе обсуждали, то я согласен со всем, только у меня еще 1 вопрос возник после прочтения:

Дает счастье всем 3 слоям населения за удовлетворение потребностей церкви : как вы представляете в игре процесс удовлетворения потребностей церкви?( это типо какое-то специальное временное задание, или еще что?)

**Comment by fshpsmgc** [comment]
>Производительность меняется со временем
В целом, неплохая идея, можно ещё подумать над тем, то лучше -- это или ограниченное количество ресурсов на точке, или может вообще не надо. Задачу они решают примерно одну и ту же, но эффект немного разный. 
И то, и другое сведёт производство в 0 (если не поставить ограничение, что meh), но при постепенном уменьшении, есть риск оставить игрока с чувством, что он никуда не продвигается. Например, у нас есть один завод, он производит 5 кирпичей. Через год мы захватываем ещё один завод, но он уже производит всего 3 кирпича, то есть у нас прибыли в 6 кирпичей. Ещё через год, мы захватываем ещё один завод, но они все производят по одному кирпичу, итого мы остались с меньшим количеством, чем стартовали. А потребности только растут. (Абстракция и всё такое, не придирайтесь к цифрам)

**Comment by fshpsmgc** [comment]
Фишку мира надо немного забалансить и будет заебись. Меня там смутили несколько фраз

>Бесконечный мешок ресурсов

>ресурсы определяет лор

>Не нужны караваны для торговли

Первое и второе это примерно одна проблема и она довольно просто решается. Пусть у них будут только базовые начальные ресурсы. Они нужны всегда, они не внесут дисбаланса, если будут покупаться за деньги, а не строиться, но это привнесёт уникальный стиль игры за эту сторону. Налаживание добычи этих ресурсов будет важной частью первых ~20 лет игры для всех остальных, но эта сторона вместо развития инфраструктуры будет должна развивать экономику и очень аккуратно тратить деньги, так как без денег и массовой закупки этих ресурсов ей придёт пиздец. 

Вторая проблема -- это караваны, и это довольно спорный момент. С одной стороны, это выделит эту фичу, а не сделает её обычной торговлей, но немного лучше, но с другой у нас на караванах вся игра строится, и короче хз, у меня _пока_ нет нормальных аргументов против

**Comment by aldopain** [comment]
1. Я не хочу делать иссякаемые месторождения, т.к. слишком сильно ограничивает игру. Во всяких ситибилдерах аля Тропико это работает хорошо, но, имхо, не здесь. 
2. Сведет не к нулю, а к lim(F(x)) при x -> inf. Это весьма важный момент, дающий неплохую гибкость. 
3. Да, неправильно сформулировал, не бесконечный мешок ресурсов. Если вероятность появления ресурса в каждый месяц = 0.5, а количество = от 10 до 30, то ожидание = 0.5 * (30 - 10) = 10 единиц в месяц. При t -> inf это будет inf (именно это я имел ввиду), но у нас t от 0 до конца игры. 
4. Лор скажет, какие ресурсы могут быть у магов, мы забалансим.

**Comment by fshpsmgc** [comment]
1. Может быть, да. Но с другой стороны, мы точно знаем время, сколько у нас длится игра, поэтому _может быть_ сможем забалансить так, что у нас к концу игры останется несколько самых больших шахт, за которые у нас начнётся война. (Хотя, игрок, которые остался в игре, но не контролирует одну из шахт, скорее всего в этот момент проиграет, и я не уверен, хотим ли мы этого)

2. Ну посмотрим, да. 

3. Да меня не столько смущает то, что он бесконечный, смотря какие ресурсы там будут. Но учитывая, что ты со мной согласен, что там должны быть исключительно базовые ресурсы, то норм

4. Я _очень надеюсь_, что маги умеют добывать дерево, железо, камень и т.д.

И ты как-то пропустил мою главную проблему с этой системой

>Например, у нас есть один завод, он производит 5 кирпичей. Через год мы захватываем ещё один завод, но он уже производит всего 3 кирпича, то есть у нас прибыли в 6 кирпичей. Ещё через год, мы захватываем ещё один завод, но они все производят по одному кирпичу, итого мы остались с меньшим количеством, чем стартовали. А потребности только растут. 

Потребности растут, а производство падает. Окей, да, у нас будет один игрок, который будет захватывать точки быстрее, чем у него будет падать производство, но, во-первых, это довольно сложно, а во-вторых остальные три рискуют остаться с lim(F(x)); x -> 0 ресурсами, то есть, потенциально, с количеством, которое меньше, чем в начале игры. Но при этом, подразумевается, что у него к этому моменту будет налажено производство базовых ресурсов, так как он нужны для всех последующих рецептов

**Comment by aldopain** [comment]
Так, еще раз, ты не понимаешь. 
Напомню, что x - это количество _уже_ добытых в шахте ресурсов. 
Берем функцию F(x), у которой предел не стремится к нулю и всё. С помощью этого предела + скорости функции и балансим. 
Поэтому я и хочу такую реализацию, она очень гибкая + с помощью нее и лимит можно настроить.

**Comment by aldopain** [comment]
> Дает счастье всем 3 слоям населения за удовлетворение потребностей церкви : как вы представляете в игре процесс удовлетворения потребностей церкви?( это типо какое-то специальное временное задание, или еще что?)

Нет, условно говоря церковь нуждается в 1 тонне серебра каждый месяц. 
Привозишь в город - церковь довольна, работает и горожане счастливы. 
Не привозишь - батюшка серчает, прихожан выгоняет, горожане в депрессии.
